api/src/main/java/org/apache/iceberg/Transaction.java

```diff
@@ -26,6 +26,14 @@
  * A transaction for performing multiple updates to a table.
  */
 public interface Transaction {
+
+  enum TransactionType {
```

Comment:
I don't think it is necessary to move this to the public API since both Transactions and BaseTransaction are in core.